src/DataProtection/DataProtection/src/KeyManagement/KeyRingProvider.cs

```diff
@@ -66,31 +66,54 @@ private CacheableKeyRing CreateCacheableKeyRingCore(DateTimeOffset now, IKey? ke
 
         // Fetch the current default key from the list of all keys
         var defaultKeyPolicy = _defaultKeyResolver.ResolveDefaultKeyPolicy(now, allKeys);
-        if (!defaultKeyPolicy.ShouldGenerateNewKey)
-        {
-            CryptoUtil.Assert(defaultKeyPolicy.DefaultKey != null, "Expected to see a default key.");
-            return CreateCacheableKeyRingCoreStep2(now, cacheExpirationToken, defaultKeyPolicy.DefaultKey, allKeys);
-        }
-
-        _logger.PolicyResolutionStatesThatANewKeyShouldBeAddedToTheKeyRing();
+        var defaultKey = defaultKeyPolicy.DefaultKey;
 
         // We shouldn't call CreateKey more than once, else we risk stack diving. This code path shouldn't
```

Comment:
The comment about this being rare assumes `ShouldGenerateNewKey` is false, which is no longer the case.  The behavior should be equivalent (since `defaultKey` will still win), but the comment ought to be updated.